src/Adapter/Product/CommandHandler/UpdateProductOptionsHandler.php

```diff
@@ -107,17 +107,21 @@ private function fillUpdatableProperties(Product $product, UpdateProductOptionsC
             $product->available_for_order = $command->isAvailableForOrder();
             $updatableProperties[] = 'available_for_order';
         }
+        $availableForOrder = $product->available_for_order;
+
+        if (null !== $command->showPrice() && $availableForOrder) {
```

Comment:
With @MeKeyCool we spent a few minutes understanding that these two conditions actually cover all cases, the default behavior being implicit, a comment would be nice